packages/admin/admin-rte/src/core/translation/ToolbarButton.tsx

```diff
@@ -22,15 +27,32 @@ function ToolbarButton({ editorState, setEditorState, options }: IControlProps):
 
         const translatedEditorState = htmlToState({ html: translation, entities });
 
-        setEditorState(translatedEditorState);
+        if (translationContext.showDialog) {
+            setTranslationEditorState(translatedEditorState);
+            setOpen(true);
+        } else {
+            setEditorState(translatedEditorState);
+        }
     }
 
     return (
-        <Tooltip title={<FormattedMessage id="comet.rte.translation.buttonTooltip" defaultMessage="Translate" />} placement="top">
-            <span>
-                <ControlButton icon={Translate} onButtonClick={handleClick} />
-            </span>
-        </Tooltip>
+        <>
+            <Tooltip title={<FormattedMessage id="comet.rte.translation.buttonTooltip" defaultMessage="Translate" />} placement="top">
+                <span>
+                    <ControlButton icon={Translate} onButtonClick={handleClick} />
+                </span>
+            </Tooltip>
+            {translationEditorState && (
```

Comment:
It's odd that you have `open` but check for translationEditorState here

Reply:
Changed it to  `{pendingEditorState && translationEditorState && (`

I kept the check for translationEditorState cause it would require setting the editorState optional and ` if (!translatedText) return null; `. Should be better to handle this outside the DialogComponent and should not mount the dialog component if it does not have the required data.